packages/create-cloudflare/src/frameworks/next/index.ts

```diff
@@ -22,34 +28,58 @@ const generate = async (ctx: PagesGeneratorContext) => {
 	);
 };
 
+const getApiTemplate = (
+	apiPath: string,
+	isTypescript: boolean
+): [string, string] => {
+	const isAppDir = /\/app\/api$/.test(apiPath);
+
+	if (isAppDir) {
+		// App directory uses route handlers that are defined in a subdirectory (`/api/hello/route.ts`).
+		const routeHandlerPath = `${apiPath}/hello`;
+		mkdirSync(routeHandlerPath, { recursive: true });
+
+		return isTypescript
+			? [`${routeHandlerPath}/route.ts`, apiAppDirHelloTs]
+			: [`${routeHandlerPath}/route.js`, apiAppDirHelloJs];
+	}
+
+	return isTypescript
+		? [`${apiPath}/hello.ts`, apiPagesDirHelloTs]
+		: [`${apiPath}/hello.js`, apiPagesDirHelloJs];
+};
+
 const configure = async (ctx: PagesGeneratorContext) => {
 	const projectName = ctx.project.name;
 
 	// Add a compatible function handler example
-	const apiPath = probePaths(
+	let apiPath = probePaths(
 		[
 			`${projectName}/pages/api`,
 			`${projectName}/src/pages/api`,
 			`${projectName}/src/app/api`,
 			`${projectName}/app/api`,
+			`${projectName}/src/app`,
+			`${projectName}/app`,
 		],
 		"Could not find the `/api` directory"
 	);
```

Comment:
If you're not checking for an `api` directory I think you need to update accordingly the variable name and the error message, no?

Reply:
modified the message and variable name